test/e2e/apps/deployment.go

```diff
@@ -1442,6 +1453,112 @@ func testRollingUpdateDeploymentWithLocalTrafficLoadBalancer(f *framework.Framew
 	}
 }
 
+func testScaleDeploymentWithLoadBalancer(f *framework.Framework) {
+	ns := f.Namespace.Name
+	c := f.ClientSet
+
+	name := "test-scale-with-lb"
+	podLabels := map[string]string{"name": name}
+	initialReplicas := int32(2)
+
+	framework.Logf("Creating Deployment %q", name)
+	d := e2edeployment.NewDeployment(name, initialReplicas, podLabels, AgnhostImageName, AgnhostImage, appsv1.RollingUpdateDeploymentStrategyType)
```

Comment:
Using a DaemonSet here might provide better coverage -- specifically the case where a node goes from 1 -> 0 endpoints. If you use a Deployment, the rolling update might happen in a single node where the node health check never fails.